### PR TITLE
removed duplicate_quantize_dynamic_node

### PR DESCRIPTION
**Pull request by danrecoskie**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #83611
* __->__ #83459

Differential Revision: [D38841053](https://our.internmc.facebook.com/intern/diff/D38841053)

### CODE REVIEW

**Review by jerryzh168 — APPROVED** [review]
thanks! changes looks good

File: torch/ao/quantization/fx/_lower_to_native_backend.py [hunk 1]
```diff
@@ -566,8 +563,11 @@ def _lower_dynamic_weighted_ref_module(model: QuantizedGraphModule):
         # remove q - dq node
         dq_node.replace_all_uses_with(input_dynamic_q_node)
         model.graph.erase_node(dq_node)
-        input_dynamic_q_node.replace_all_uses_with(input_dynamic_q_node.args[0])
-        model.graph.erase_node(input_dynamic_q_node)
+        dynamic_remove_set.add(input_dynamic_q_node)
+
+    for node in dynamic_remove_set:
+        node.replace_all_uses_with(node.args[0])
+        model.graph.erase_node(node)
```
Comment (jerryzh168) [inline torch/ao/quantization/fx/_lower_to_native_backend.py:570]: we can rely on dead code elimination to remove this I think: https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/fx/_lower_to_native_backend.py#L966

**Review by jerryzh168 — CHANGES_REQUESTED** [review]
actually I think there are more changes that's needed, for pattern:
quantize_dynamic - dequantize - fp_op
we also need to allow dequantize to be used by multiple ops, like

quantize_dynamic - dequantize - fp_op
                                           \ -- fp_op2

right now there is still a check in L532 saying dq should not be used by multiple users and in L564 
```dq_node.replace_all_uses_with(input_dynamic_q_node)``` also assumes that dq is only used by one node I think

**Review by jerryzh168 — APPROVED** [review]
lg, thanks!

**Comment by danrecoskie** [comment]
@danrecoskie has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D38841053).

**Pull request closed without merging**